DOCSUP-3908: mannWitneyUTest, welchTTest and studentTTest documentation (#19784)
ClickHouse/ClickHouse: docs/en/sql-reference/aggregate-functions/reference/mannwhitneyutest.md

---
toc_priority: 310
toc_title: mannWhitneyUTest
---

> ## mannWhitneyUTest {#mannwhitneyutest}

Reviewer: ```suggestion
# mannWhitneyUTest {#mannwhitneyutest}
```

> Applies the Mann-Whitney rank test to independent samples from distinct distributions.

Reviewer: ```suggestion
Applies the [Mann–Whitney U test](https://en.wikipedia.org/wiki/Mann%E2%80%93Whitney_U_test) to independent samples from distinct distributions.
```

> 

Reviewer: ```suggestion
Applies the [Mann–Whitney U test](https://en.wikipedia.org/wiki/Mann%E2%80%93Whitney_U_test) to independent samples from distinct distributions.
```

**Syntax**

``` sql
mannWhitneyUTest(alternative)(sample_data, sample_index)
```
Values X and Y are randomly selected from two populations. The null hypothesis is that the probability of X being greater than Y is equal to the probability of Y being greater than X. Also one-sided hypothesis can be tested.
  
Values of both samples are in the `sample_data` column. If `sample_index` equals to 0 then the value in that row belongs to the first sample. Otherwise it belongs to the second sample. 

> 

Reviewer: What are X and Y here is not clear.

**Parameters**

> -   `alternative` — alternative hypothesis. [String](../../../sql-reference/data-types/string.md)

Reviewer: ```suggestion
-   `alternative` — hypothesis tested. [String](../../../sql-reference/data-types/string.md). Possible values:
    -   `'two-sided'` (by default)
    -   `'greater'`
    -   `'less'`
```

    -   `"two-sided"`;
    -   `"greater"`;
    -   `"less"`;

>     -   if parameter is not specified then default two sided hypothesis is tested.

Reviewer: ```suggestion
-   `alternative` — hypothesis tested. [String](../../../sql-reference/data-types/string.md). Possible values:
    -   `"two-sided"` (by default)
    -   `"greater"`
    -   `"less"`
```

-   `sample_data` — sample data. [Float64](../../../sql-reference/data-types/float.md).

> -   `sample_index` — sample index. [Int8](../../../sql-reference/data-types/int-uint.md).

Reviewer: ```suggestion
-   `sample_index` — sample index. [UInt64](../../../sql-reference/data-types/int-uint.md).
```

**Returned values**

> -   statistic - calculated statistic;

Reviewer: ```suggestion
-  ` statistic` - calculated statistic;
```

> -   p-value - calculated p-value.

Reviewer: ```suggestion
-  ` statistic` - calculated statistic;
-   `p-value` - calculated p-value.
```

Type: [Float64](../../../sql-reference/data-types/float.md).

**Example**

Input table:

``` text
┌─sample_data─┬─sample_index─┐
│          10 │            0 │
│          11 │            0 │
│          12 │            0 │
│           1 │            1 │
│           2 │            1 │
│           3 │            1 │
└─────────────┴──────────────┘
```

Query:

``` sql
SELECT mannWhitneyUTest('greater')(sample_data, sample_index)
FROM mww_ttest;
```

> 

Reviewer: ``` sql
SELECT mannWhitneyUTest('greater')(sample_data, sample_index) FROM mww_ttest;
```
